### ISSUE TITLE
PVC pending Failed to get GCE GCECloudProvider with error 

### ISSUE THREAD

**Issue by IronCore864** [kubernetes/kubernetes#66564]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
--cloud-provider=gce for kubelet
service account that has permission to create gce pd added to the VM
install k8s cluster using kubeadm in the VM
Created a Dynamic StorageClass slow using provisioner: kubernetes.io/gce-pd in k8s cluster, according to official doc example:
https://kubernetes.io/docs/concepts/storage/storage-classes/#gce

Create a PVC bind with StorageClass Slow

error:
Failed to provision volume with StorageClass "slow": Failed to get GCE GCECloudProvider with error

**What you expected to happen**:
PVC is created correctly and bind with StorageClass Slow

**How to reproduce it (as minimally and precisely as possible)**:
- kubeadm install 1.11.1 in gcloud VM, single node
- create service account in gcloud that has permission to create gce pd, and grant the permission to the VM
- set --cloud-provider to gce for kubelet in KUBELET_KUBECONFIG_ARGS in /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
- set /etc/kubernetes/cloud-config with content:
[Global]
project-id = "xxx"

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.11.1
- Cloud provider or hardware configuration: gce
- OS (e.g. from /etc/os-release): Ubuntu 16.04.4 LTS
- Kernel (e.g. `uname -a`): 4.13.0-1019-gcp #23-Ubuntu SMP
- Install tools: kubeadm
- Others:

Basically same issue as https://stackoverflow.com/questions/37421540/container-vm-image-with-gpd-volumes-fails-with-failed-to-get-gce-cloud-provider but that one weirdly is closed because some people thought it was local storage while it's not.

**Comment by msau42** [comment]
@neolit123 kube-controller-manager is the process that handles provisioning of volumes, so that needs the cloud provider passed in

**Comment by neolit123** [comment]
@IronCore864, as @msau42 suggests, try passing `cloud-provider: gce` to the controller manager from the kubeadm config as explained here:

https://kubernetes.io/docs/setup/independent/control-plane-flags/#controllermanager-flags

might as well pass it to the api server as well.

**Comment by IronCore864** [comment]
I'll try this asap and get back later.

**Comment by IronCore864** [comment]
Works, great. After passing cloud provider to both controller manager and api server. Please go ahead and close this ticket but I really appreciate it if we could improve the doc to be more detailed on this 👍

**Comment by neolit123** [comment]
ref: https://github.com/kubernetes/kubernetes/issues/66564

**Comment by kigawas** [comment]
For how to upgrade kubeadm config (k8s version 1.14.2):

Say there is a file `gce.yml`:
```yaml
apiVersion: kubeadm.k8s.io/v1beta1
kind: ClusterConfiguration
controllerManager:
  extraArgs:
    cloud-provider: gce
```
And `sudo kubeadm upgrade apply --config gce.yml`
